packages/patternfly-4/react-core/src/components/Nav/NavListItem.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** Content rendered inside the list item */
+  children: PropTypes.node.isRequired,
+  /** Additional classes added to the list item */
+  className: PropTypes.string,
+  /** Target navigation link */
+  to: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  className: ''
+};
+
+/* eslint react/prop-types: 0 */
+class NavListItem extends React.Component {
+  onUpdate(e) {
+    const { updateActive, groupId, id } = this.props;
+    if (e) {
+      e.stopPropagation();
+    }
+    // Callback to Nav
+    updateActive(groupId, id);
+  }
+
+  render() {
+    const { className, children, to, id, groupId, activeGroup, activeItem, updateActive, ...props } = this.props;
+    const defaultLink = (
+      <a
+        href={to}
+        onClick={e => this.onUpdate(e)}
+        className={css(
+          styles.navLink,
+          activeGroup === groupId && activeItem === id && styles.modifiers.current,
+          className
+        )}
+        aria-current="page"
+      >
+        {children}
+      </a>
+    );
+    const reactElement = React.isValidElement(children);
+    let clonedChild;
+    if (reactElement) {
+      // This could be a react-router Link for example
```

Comment:
`onUpdate` function should handle all these things and you shouldn't have to clone the children and such.

Reply:
I need to clone here to attach the update function to a react element